include/deal.II/lac/arpack_solver.h

```diff
@@ -482,9 +497,13 @@ void ArpackSolver::solve (
           Assert (false, ExcArpackInfodneupd(info));
         }
 
-      for (size_type i=0; i<eigenvectors.size(); ++i)
+      const unsigned int n_eigenvecs = eigenvectors.size() / 2;
+      for (size_type i=0; i<n_eigenvecs; ++i)
         for (unsigned int j=0; j<n; ++j)
-          eigenvectors[i](j) = v[i*n+j];
+          {
+            eigenvectors[i](j) = z[2*i*n+j];
+            eigenvectors[i+n_eigenvecs](j) = z[(2*i+1)*n+j];
```

Comment:
If eigenvectors happens to be of odd size this goes out of bounds, doesn't it?

While we're already fixing the implementation with this very pull-request - what about bringing some sense to the interface by clearly stating what the requirements for <code>eigenvectors</code> in the <code>solve</code> function are? And do some bounds checking for it? (Or have I missed something?)

E.g. at some point we do
```
VECTOR src,dst,tmp;
src.reinit(eigenvectors[0]);
dst.reinit(src);
tmp.reinit(src);
```
This requires <code>eigenvectors</code> to be at least partially set up.